compiler/fir/raw-fir/raw-fir.common/src/org/jetbrains/kotlin/fir/builder/BaseFirBuilder.kt

```diff
@@ -214,13 +214,25 @@ abstract class BaseFirBuilder<T>(val baseSession: FirSession, val context: Conte
         }
     }
 
-    fun FirLoopBuilder.configure(generateBlock: () -> FirBlock): FirLoop {
+    fun FirLoopBuilder.prepareTarget() {
```

Comment:
Can you think of a way to make sure there is a single call to `prepareTarget` for every call to `buildAndBind` (and vice versa)? It might be overkill but could prevent errors down the road, just in case.

Reply:
It's not a perfect way, but `firLoopTargets.removeLast` in `configure` will be broken if `prepareTarget` and `configure` are not matched.

Reply:
Actually, we can store the loop target generated at `prepareTarget`, and make sure `configure` is seeing the same target on the target stack top.